src/raw_list.rs

```diff
@@ -0,0 +1,562 @@
+        }
+    }
+
+    /// Returns the element the cursor is currently positioned on.
+    pub(crate) fn current(&self) -> Option<&'a G::EntryType> {
+        let cur = self.cursor.cur?;
+        // SAFETY: Objects must be kept alive while on the list.
+        Some(unsafe { &*cur.as_ptr() })
+    }
+
+    /// Moves the cursor to the next element.
+    pub(crate) fn move_next(&mut self) {
+        self.cursor.move_next(self.list);
+    }
+
+    /// Moves the cursor to the prev element.
+    #[allow(dead_code)]
+    pub(crate) fn move_prev(&mut self) {
+        self.cursor.move_prev(self.list);
+    }
+}
+
+pub(crate) struct CursorMut<'a, G: GetLinks> {
+    cursor: CommonCursor<G>,
+    list: &'a mut RawList<G>,
+}
+
+impl<'a, G: GetLinks> CursorMut<'a, G> {
+    fn new(list: &'a mut RawList<G>, cur: Option<NonNull<G::EntryType>>) -> Self {
+        Self {
+            list,
+            cursor: CommonCursor::new(cur),
+        }
+    }
+
+    pub(crate) fn current(&mut self) -> Option<&mut G::EntryType> {
+        let cur = self.cursor.cur?;
+        // SAFETY: Objects must be kept alive while on the list.
+        Some(unsafe { &mut *cur.as_ptr() })
+    }
+
+    /// Removes the entry the cursor is pointing to and advances the cursor to the next entry. It
+    /// returns a raw pointer to the removed element (if one is removed).
+    pub(crate) fn remove_current(&mut self) -> Option<NonNull<G::EntryType>> {
+        let entry = self.cursor.cur?;
+        self.cursor.move_next(self.list);
+        // SAFETY: The entry is on the list as we just got it from there and it cannot change.
+        unsafe { self.list.remove(entry.as_ref()) };
+        Some(entry)
+    }
+
+    pub(crate) fn peek_next(&mut self) -> Option<&mut G::EntryType> {
+        let mut new = CommonCursor::new(self.cursor.cur);
+        new.move_next(self.list);
+        // SAFETY: Objects must be kept alive while on the list.
+        Some(unsafe { &mut *new.cur?.as_ptr() })
+    }
+
+    pub(crate) fn peek_prev(&mut self) -> Option<&mut G::EntryType> {
+        let mut new = CommonCursor::new(self.cursor.cur);
+        new.move_prev(self.list);
+        // SAFETY: Objects must be kept alive while on the list.
+        Some(unsafe { &mut *new.cur?.as_ptr() })
+    }
+
+    pub(crate) fn move_next(&mut self) {
+        self.cursor.move_next(self.list);
+    }
+
+    #[allow(dead_code)]
+    pub(crate) fn move_prev(&mut self) {
+        self.cursor.move_prev(self.list);
+    }
+}
+
+impl<'a, G: GetLinks> iter::IntoIterator for &'a RawList<G> {
+    type Item = &'a G::EntryType;
+    type IntoIter = Iterator<'a, G>;
+    fn into_iter(self) -> Self::IntoIter {
+        self.iter()
+    }
+}
+
+/// An iterator for the linked list.
+pub(crate) struct Iterator<'a, G: GetLinks> {
+    cursor_front: Cursor<'a, G>,
+    cursor_back: Cursor<'a, G>,
+}
+
+impl<'a, G: GetLinks> Iterator<'a, G> {
+    fn new(cursor_front: Cursor<'a, G>, cursor_back: Cursor<'a, G>) -> Self {
```

Comment:
const fn

Reply:
![1729477374132](https://github.com/user-attachments/assets/0042ad7c-4162-4f9d-bcef-9267500bd262)
not all funcs allowed  add const.